### PR TITLE
chore: improve error logs (#10592)

### PR DESCRIPTION
**Pull request by eogns47**
This PR is for issue #10592.

I have completed error wrapping for all files in util/helm.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by reggie-k — CHANGES_REQUESTED** [review]
I think the actual run was at RunWithRedactor, so the message "failed to get command args to log" is better
Line  74 util/helm/cmd.go

**Comment by eogns47** [comment]
> I think the actual run was at RunWithRedactor, so the message "failed to get command args to log" is better Line 74 util/helm/cmd.go

I think so, I'll fix it and re-commit!

**Comment by eogns47** [comment]
@reggie-k I think I may have signed off on someone else's commit while rebasing on my branch. 🥲

**Comment by reggie-k** [comment]
Thanks, I am also wondering about how much error info for creds/ca and other security-related and potentially sensitive stuff we wanna give, would like to hear more maintainer opinions on that.

**Comment by eogns47** [comment]
> Thanks, I am also wondering about how much error info for creds/ca and other security-related and potentially sensitive stuff we wanna give, would like to hear more maintainer opinions on that.

Thanks for the comment, I hadn't thought of that. 🥲

**Comment by eogns47** [comment]
@crenshaw-dev Hello crenshaw!
If any of the error information I provide in my PRs is a security risk, please review it. I'll fix it right away.

File: util/helm/client.go [hunk 1]
```diff
@@ -192,34 +198,34 @@ func (c *nativeHelmChart) ExtractChart(chart string, version string, project str
...
+			return "", nil, fmt.Errorf("error reading directory %s: %w", tempDest, err)
 		}
 		if len(infos) != 1 {
 			return "", nil, fmt.Errorf("expected 1 file, found %v", len(infos))
 		}
 
 		err = os.Rename(filepath.Join(tempDest, infos[0].Name()), cachedChartPath)
 		if err != nil {
-			return "", nil, err
+			return "", nil, fmt.Errorf("error renaming file from %s to %s: %w", filepath.Join(tempDest, infos[0].Name()), cachedChartPath, err)
```
Comment (rumstead) [inline util/helm/client.go:221]: Should we save `filepath.Join(tempDest, infos[0].Name())` into a variable and use it on line 219 as well?

Comment (eogns47) [reply util/helm/client.go:221]: Thank you so much. I think that's a good idea 😃

**Review by rumstead — APPROVED** [review]

**Comment by rumstead** [comment]
my approval doesnt mean much but LGTM!

**Comment by eogns47** [comment]
> my approval doesnt mean much but LGTM!

Thanks for the review though! That's very kind of you. 🥹

**Review by agaudreault — APPROVED** [review]
LGTM. I dont see any security impact because we add textual information to the already existing error object.

**Pull request merged**